classes/local/form/login_form.php

```diff
@@ -34,21 +34,12 @@ class login_form extends \moodleform {
      */
     public function definition() {
         $mform = $this->_form;
-        $factorname = $this->_customdata['factor_name'];
-        $gracemode = $this->_customdata['grace_mode'];
+        $factor = $this->_customdata['factor'];
 
-        $mform->addElement('hidden', 'factor_name', $factorname);
+        $mform->addElement('hidden', 'factor_name', $factor->name);
```

Comment:
Do we need this at all? This feels like a potential vector

Reply:
Its just to track the factor name, as the validation has to re-get the factor, as it cant be passed in as a parameter